sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -99,6 +99,14 @@ private[sql] class HiveSessionCatalog(
           throw new AnalysisException(s"No handler for Hive UDF '${clazz.getCanonicalName}'")
         }
       } catch {
+        case ise: IllegalStateException =>
```

Comment:
Hi, @wangyum .
Can we move this after line 112, `case NonFatal(e) =>`?
IMO, these 8 lines are mostly duplication logic of line 113~116. The only difference is error message.
I guess we can append the new error message more simply.